src/ui-layout.js

```diff
@@ -796,6 +807,31 @@ angular.module('ui.layout', [])
             if(!element.hasClass('stretch')) element.addClass('stretch');
             if(!element.hasClass('ui-layout-container')) element.addClass('ui-layout-container');
 
+            scope.$watch('collapsed', function (val, old) {
+              if (old !== undefined && val !== old) {
+                var index = scope.container.index;
+                var splitter = ctrl.containers[index + 1],
+                  el;
+
+                if (splitter) {
+                  el = splitter.element[0].children[0];
+                } else {
+                  splitter = ctrl.containers[index - 1];
+                  el = splitter.element[0].children[1];
+                }
+
+
+                $timeout(function(){
+                  angular.element(el).trigger('click');
+                });
+
+                $timeout(function(){
```

Comment:
In this context it looks like $timeout is being used to handle async control flow, would it be possible to rewrite this using a promise/$q. 
Other than that the PR looks good to me.